pyproject.toml

```diff
@@ -20,7 +19,10 @@ classifiers = [
     'Topic :: Software Development :: Testing',
     'Topic :: Utilities',
 ]
-dependencies = ["pytest>=7.0.0"]
+dependencies = [
+    "pytest >= 7.0.0",
+    "typing-extensions >= 4.12.2, < 5; python_version < '3.11'"
```

Comment:
For libraries like `typing-extensions` there shouldn't be a reason to upper-bound it, but happy to learn a counter-reason. As for the lower-bound, any reasoning for that? Usually I go with unbounded until I find a reason for a bound and document it.